Timezone date

Raymoz101 opened the pull request:
- Fixed the speedtest result date appearing in en-US format by grabbing the environment Timezone and setting the Date to that.
- The "toLocaleDateString" was too specific and only returns the Date. "toLocaleString" returns time as well as date, then removed 2 x zeros to stick with the tidy output.
- Added the suggestion to include the Timezone in the run command of the Docker image to the readme.

Raymoz101 committed:
replaced timenow function with en-AU localestring

Raymoz101 committed:
updated to remove leading zeroes on day and month

Raymoz101 committed:
improved the date string to get the system timezone and print based off that

Raymoz101 committed:
Update README.md

added the option of specifying timezone in the docker run command, only required if Timezone not specified during docker image build

Raymoz101 commented:
Current master build:
![abb-speedtest-cli_Before](https://user-images.githubusercontent.com/6777115/92318942-7c362b80-f056-11ea-95d3-f8dc25ecbf9f.png)
Timezone-date build:
![abb-speedtest-cli_After](https://user-images.githubusercontent.com/6777115/92318946-7fc9b280-f056-11ea-9dc7-b290745fe0b6.png)

TJC commented on README.md:
```diff
@@ -80,6 +80,8 @@ Then to run it:
 # Optionally, include parameters like this:
 docker run --rm abb-speedtest --json
+# You can also set the Docker container timezone like this:
+docker run --rm -e TZ=Australia/Sydney abb-speedtest
```
It might be good to set that timezone as the default in the Docker container.
People in WA, QLD, NT and SA will still want to override it, but I think using a default that covers 40% of the population of the country is better than it using a default that covers 0% percent :)

Raymoz101 replied:
Done.
Added that to the Dockerfile, and this triggered an auto-build, I pulled that and ran without specifying the -e TZ in the run command and the result date was accurate.
I linked my fork to a Docker hub repo and setup auto-builds, so each time I pull @lukealford updates it will update the docker builds.
Should also mean it will start to show up for people using Synology NAS etc.
https://hub.docker.com/repository/docker/raymoz101/abb-speedtest-cli

Raymoz101 committed:
Updated Dockerfile to set Australia/Sydney as default Timezone

Raymoz101 committed:
Update README.md

Clarified that the timezone is preset during the Docker build via the Dockerfile and explained how to manually adjust. Will look into ways of passing timezone safely into the image.

TJC commented on README.md:
````diff
@@ -74,14 +74,21 @@ Instead of installing all components directly, you can also build a Docker conta
 ```
+Note: Australia/Sydney is defined in the Dockerfile as the default Timezone.
+
+Add the following environment variable to your run command (Replacing STATE with your location) to specify an alternate Timezone.
````
When you say location.... I think what you mean is "one of the allowed zone definitions from `/usr/share/zoneinfo/Australia` in Ubuntu LTS 18".

It's a bit odd in there, because the names are not consistent.
These states are listed: ACT, NSW, Queensland, Tasmania, Victoria. (ACT and NSW are abbreviated, the others are not)
But there's no SA, NT, or WA. They're actually in there as "South", "North" and "West".

I believe the best-practice usage is to refer to the capital cities, eg. Sydney, Melbourne, Darwin, Adelaide, Canberra, Perth, Hobart, Brisbane.

And then the more distant places don't follow any convention, like Broken Hill, Yancowinna, LHI (Lord Howe island), Currie (King island), etc

TJC replied:
I guess people will figure it out, but maybe say `Replacing Sydney with your local capital city name` instead of state?

Raymoz101 replied:
Good point. 

Agreed. I'll re-word it tonight to be clearer. 

I've seen examples where people mount an /etc/zones path as a Read-Only volume to pass system Timezone into the container, removing the need for specifying. I might test it out. But also being based on Alpine it might not have tzdata included, which is a requirement for that to work.

Raymoz101 committed:
Update Dockerfile

Defined ARG "LOCALEZONE"
--no-cache add curl and hit worldtime api
grep out the users timezone
pass this into ENV TZ

Raymoz101 committed:
Update Dockerfile

Workaround to pass timezone to local file and then set it as the image timezone using tzdata

Raymoz101 committed:
Update README.md

Timezone is now dynamic during build, tidying up readme to suit

Raymoz101 committed:
Merge pull request #1 from Raymoz101/Timezone-Dockerfile

Timezone dockerfile

Raymoz101 commented on Dockerfile:
```diff
@@ -11,6 +11,12 @@ RUN apk update && \
     mkdir -p /app
 COPY . /app/
 WORKDIR /app
+RUN apk --no-cache add grep && \
+	curl --silent "http://worldtimeapi.org/api/ip" --stderr - | grep -oP '(?<=timezone":").*(?=","unixtime)' > /app/timezone
+RUN apk add tzdata && \
+	LOCALEZONE=$(cat /app/timezone) && \
+	ln -snf /usr/share/zoneinfo/$LOCALEZONE /etc/localtime && \
+	echo $LOCALEZONE > /etc/timezone
```
Not sure why I had so much fun getting this to work.
Let me know your thoughts. Basically this command "curl --silent "http://worldtimeapi.org/api/ip" --stderr - | grep -oP '(?<=timezone":").*(?=","unixtime)'" hits an API for a JSON result and greps it down to just your timezone. 
Getting this into the Docker image itself isn't straight forward. But done.

TJC replied:
This looks like it sets the timezone at the time of _building_ the image, right?

So, if the Docker image is getting built by Docker Hub, in their US AWS account, then.. won't that set the default timezone to be US Eastern Time?

TJC replied:
I feel like we'd be better off with an Australian default timezone rather than US..
If we're going to try and calculate it on the fly, then at least it should be done at the time of _running_ the container, not building it?

lukealford replied:
I'd default to an AU personally and offset from there this way you never get a timezone outside of AU. But for now, I've merged it
